js/ui/text_box/ui.text_editor.base.js

```diff
@@ -438,6 +448,45 @@ const TextEditorBase = Editor.inherit({
         this._input().prop('spellcheck', this.option('spellcheck'));
     },
 
+    _renderLabel: function(container) {
+        const TEXTEDITOR_WITH_AFTER_BUTTONS_CLASS = 'dx-texteditor-with-after-buttons';
+        const $container = container || this.$element();
+
+        if(!this.label && $container.find('.' + TEXTEDITOR_LABEL_CLASS).length === 1 || $container.find('.' + TEXTEDITOR_LABEL_CLASS).length === 2) {
+            this.$element().find('.' + TEXTEDITOR_LABEL_CLASS).first().remove();
+        }
+
+        if(this._$label) {
+            this._$label.remove();
+            this._$label = null;
+        }
+
+        this.$element()
+            .removeClass(TEXTEDITOR_WITH_LABEL_CLASS)
+            .removeClass(TEXTEDITOR_WITH_FLOATING_LABEL_CLASS)
+            .removeClass(TEXTEDITOR_WITH_AFTER_BUTTONS_CLASS);
+
+        if(!this.option('label') || this.option('labelMode') === 'hidden') return;
+
+        this.$element().addClass(this.option('labelMode') === 'floating' ? TEXTEDITOR_WITH_FLOATING_LABEL_CLASS : TEXTEDITOR_WITH_LABEL_CLASS);
+
+        const labelText = this.option('label');
+        const $label = this._$label = $('<div>')
+            .addClass(TEXTEDITOR_LABEL_CLASS)
+            .html('<span>' + labelText + '</span>');
+
+        const $textEditorInputContainer = $container.find('.' + TEXTEDITOR_INPUT_CONTAINER_CLASS);
+
+        this.NAME === 'dxLookup' ?
+            $label.appendTo(container) :
+            $label.appendTo($textEditorInputContainer);
```

Comment:
Why not extract the label as a separate component to which to pass the required parameters?

Using a descendant component in a base component doesn't look good